src/agent/src/storage/block_handler.rs

```diff
@@ -28,8 +29,48 @@ use crate::device::block_device_handler::{
 };
 use crate::device::nvdimm_device_handler::wait_for_pmem_device;
 use crate::device::scsi_device_handler::get_scsi_device_name;
-use crate::pci;
 use crate::storage::{common_storage_handler, new_device, StorageContext, StorageHandler};
+use crate::{confidential_data_hub, pci, AGENT_CONFIG};
+use slog::Logger;
+
+fn get_device_number(dev_path: &str, metadata: Option<&fs::Metadata>) -> Result<String> {
+    let dev_id = match metadata {
+        Some(m) => m.rdev(),
+        None => {
+            let m =
+                fs::metadata(dev_path).context(format!("get metadata on file {:?}", dev_path))?;
+            m.rdev()
+        }
+    };
+    Ok(format!("{}:{}", major(dev_id), minor(dev_id)))
+}
+
+async fn handle_block_storage(
+    logger: &Logger,
+    storage: &Storage,
+    dev_num: &str,
+) -> Result<Arc<dyn StorageDevice>> {
+    let has_ephemeral_encryption = storage
+        .driver_options
+        .contains(&"encryption_key=ephemeral".to_string());
+
+    if has_ephemeral_encryption {
+        let integrity = AGENT_CONFIG.secure_storage_integrity.to_string();
+        let options = std::collections::HashMap::from([
+            ("deviceId".to_string(), dev_num.to_string()),
+            ("encryptType".to_string(), "LUKS".to_string()),
+            ("dataIntegrity".to_string(), integrity),
+        ]);
+
+        confidential_data_hub::secure_mount("BlockDevice", &options, vec![], &storage.mount_point)
+            .await?;
+
+        new_device(storage.mount_point.clone())
+    } else {
+        let path = common_storage_handler(logger, storage)?;
+        new_device(path)
```

Comment:
Maybe not the right spot to ask, but: what if we use virtio-blk (i.e. shared_fs=none) but do not want `block-encrypted`? Looking at the general flow here and in other places, seems like yes, just want to double-check.

Reply:
If shared_fs=none and emptydir_mode=shared-fs, the emptyDir will maintain today's behavior (tmpfs).

Reply:
emptyDirs do not support unencrypted virtio-blk/scsi 🙂 This would've to be implemented separately (which I'm considering).